src/lib/viewers/media/DashViewer.js

```diff
@@ -594,15 +606,52 @@ class DashViewer extends VideoBaseViewer {
      */
     loadSubtitles() {
         // Load subtitles from video, if available
-        this.textTracks = this.player.getTextTracks().sort((track1, track2) => track1.id - track2.id);
+        this.textTracks = this.player
+            .getTextTracks()
+            .sort((track1, track2) => track1.id - track2.id)
+            .map(track => ({ ...track, displayLanguage: getLanguageName(track.language) || track.language }));
+
         if (this.textTracks.length > 0) {
-            this.mediaControls.initSubtitles(
-                this.textTracks.map(track => getLanguageName(track.language) || track.language),
-                getLanguageName(this.options.location.locale.substring(0, 2)),
-            );
+            if (this.getViewerOption('useReactControls')) {
+                this.initSubtitles();
+            } else {
+                this.mediaControls.initSubtitles(
+                    this.textTracks.map(track => getLanguageName(track.language) || track.language),
+                    getLanguageName(this.options.location.locale.substring(0, 2)),
+                );
+            }
         }
     }
 
+    /**
+     * Initializes the subtitles to the appropriate text track, if applicable
+     *
+     * @return {void}
+     */
+    initSubtitles() {
+        const cachedSubtitle = this.cache.get('media-subtitles');
+        const clientLanguage = getLanguageName(this.options.location.locale.substring(0, 2));
+
+        // Last video was watched with subtitles, so turn them on here too
+        if (cachedSubtitle !== null && cachedSubtitle !== SUBTITLES_OFF) {
+            // Do intelligent selection: Prefer user's language, fallback to English, then first subtitle in list
+            // Use the previewer's locale to determine preferred language
+            let idx = this.textTracks.findIndex(({ language }) => language === clientLanguage);
+            if (idx === -1) {
+                // Fall back to English if user's language doesn't exist
+                idx = this.textTracks.findIndex(({ language }) => language === 'English');
+                if (idx === -1) {
```

Comment:
Did we copy this logic from elsewhere? If not, can we simplify these cases and avoid the arrow-like shape the code is taking on?

Reply:
Yeah it's copied from https://github.com/box/box-content-preview/blob/6be9662f921dfe1148c2cce16e07d32373ef70cc/src/lib/viewers/media/Settings.js#L645

I can see if we can simplify it